build/patches/add-user-scripts.v5.patch

```diff
@@ -0,0 +1,9810 @@
++    return static_cast<base::DictionaryValue*>(dict_item);
++  }
++
++ private:
++  const std::string id_;
++
++  DISALLOW_COPY_AND_ASSIGN(PrefUpdate);
++};
++
++bool GetInt64FromPref(const base::DictionaryValue* dict,
++                      const std::string& key,
++                      int64_t* value) {
++  DCHECK(dict);
++  std::string value_str;
++  if (!dict->GetStringWithoutPathExpansion(key, &value_str)) {
++    VLOG(2) << "Can't find key in local pref dictionary. Invalid key: " << key
++            << ".";
++    return false;
++  }
++
++  if (!base::StringToInt64(value_str, value)) {
++    VLOG(2) << "Can't change string to int64_t. Invalid string value: "
++            << value_str << ".";
++    return false;
++  }
++
++  return true;
++}
++
++}
++
++UserScriptsPrefs::UserScriptsPrefs(
++    PrefService* prefs)
++    : prefs_(prefs) {
++}
++
++// static
++void UserScriptsPrefs::RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
++  registry->RegisterBooleanPref(kUserScriptsEnabled, false);
++  registry->RegisterDictionaryPref(kUserScriptsList);
++}
++
++bool UserScriptsPrefs::IsEnabled() {
++  return prefs_->GetBoolean(kUserScriptsEnabled);
++}
++
++void UserScriptsPrefs::SetEnabled(bool enabled) {
++  prefs_->SetBoolean(kUserScriptsEnabled, enabled);
++}
++
++void UserScriptsPrefs::CompareWithPrefs(UserScriptList& user_scripts) {
++  if( IsEnabled() == false ) {
++    LOG(INFO) << "UserScripts disabled";
++    user_scripts.clear();
++    return;
++  }
++
++  std::vector<std::string> all_scripts;
++
++  auto it = user_scripts.begin();
++  while (it != user_scripts.end())
++  {
++    std::string key = it->get()->key();
++    all_scripts.push_back(key);
++
++    std::unique_ptr<UserScriptsListPrefs::ScriptInfo> scriptInfo =
++        UserScriptsPrefs::CreateScriptInfoFromPrefs(key);
++
++    // add or update prefs
++    scriptInfo->set_name(it->get()->name());
++    scriptInfo->set_description(it->get()->description());
++    scriptInfo->set_version(it->get()->version());
++
++    // LOG(INFO) << "---UserScriptsPrefs::CompareWithPrefs name " << scriptInfo->name();
++
++    PrefUpdate update(prefs_, key, kUserScriptsList);
++    base::DictionaryValue* script_dict = update.Get();
++
++    script_dict->SetString(kScriptName, scriptInfo->name());
++    script_dict->SetString(kScriptDescription, scriptInfo->description());
++    script_dict->SetBoolean(kScriptIsEnabled, scriptInfo->enabled);
++    script_dict->SetString(kScriptVersion, scriptInfo->version());
++
++    std::string install_time_str =
++        base::NumberToString(scriptInfo->install_time.ToInternalValue());
++    script_dict->SetString(kScriptInstallTime, install_time_str);
++
++   if (!scriptInfo->enabled) {
++      // LOG(INFO) << "---UserScriptsPrefs::CompareWithPrefs removed " << key;
++      it = user_scripts.erase(it);
++    } else {
++      // LOG(INFO) << "---UserScriptsPrefs::CompareWithPrefs using " << key;
```

Comment:
Please uncomment or delete this line.

Reply:
fixed in https://github.com/bromite/bromite/commit/2812a0f9a76a7b3052b7a408344095680e7244ff